calculator.c

```diff
@@ -0,0 +1,124 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int numberStack[50];
+int numberStackTop = -1;
+int operatorStack[50];
+int operatorStackTop = -1;
+
+int evaluateExpression(char expression[]);
+int processNumber(const char expression[], int index);
+void processOperator(char character);
+void calculate();
+int operatorPrecedence(char operator);
+int isDigit(char character);
+int isSpace(char character);
+void errorHandler(const char *errorMessage);
+
+int main() {
+  char expression[80];
```

Comment:
Was it specified in the assignment that the string input will have max length of 80?

Reply:
No, it is not specified in the assignment that the string input will have max length of 80.
In place of fixed length input i have used dynamic memory allocation using `malloac`.